lib/nx_gzlib.c

```diff
@@ -185,14 +203,33 @@ gzFile nx_gzopen_(const char* path, int fd, const char *mode)
 	return file;
 }
 
-int nx_gzread(gzFile file, void *buf, unsigned len)
+gzFile nx_gzopen_(const char* path, int fd, const char *mode) {
+	return __gzopen_(path, fd, mode, 1);
+}
+
+gzFile nx_gzdopen(int fd, const char *mode)
+{
+	return __gzopen_(NULL, fd, mode, 1);
+}
+
+gzFile nx_gzopen(const char* path, const char *mode)
+{
+	return __gzopen_(path, 0, mode, 1);
+}
+
+int __gzread(gzFile file, void *buf, unsigned len, int force_nx)
 {
 	gz_statep state = (gz_statep) file;
 	z_streamp stream = &(state->strm);
 	int ret, err;
 	const uInt max = len > 10? 10 : 1;
 	uLong last_total_out = stream->total_out;
+	typeof(inflate) (*inf);
 
+	if (force_nx)
+		inf = &nx_inflate;
+	else
+		inf = &inflate;
```

Comment:
`__gzclose` has multiple calls to other functions, I think it would be cleaner if you set the function pointers like here.